### ISSUE TITLE
Add the google-cloud-java BOM to pom.xml

### ISSUE THREAD

**Issue by elharo** [GoogleCloudPlatform/appengine-plugins-core#584]
and use it for mutual dependencies

**Comment by patflynn** [comment]
do we use google-cloud-java from appengine-plugins-core?

**Comment by elharo** [comment]
Not really, but we do use some of the same dependencies; and we should experiment with what adding a BOM is likely to mean in practice. Until we have a more general open source cloud BOM, google-cloud-java is a good place to start. If this doesn't work, then that will tell us some things we're going to need to fix in any BOM we do create.

**Comment by elharo** [comment]
Looks like the BOM doesn't pull in transitive dependencies so there actually isn't any overlap. Good to know.

**Comment by elharo** [comment]
Adding google-cloud-java's BOM doesn't help. Second idea: can/should we move the dependencies of this library into a BOM that could be the seed for a more general BOM?
